keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ArtifactController.kt

```diff
@@ -107,6 +108,42 @@ class ArtifactController(
     artifactRepository.deletePin(deliveryConfig, targetEnvironment)
   }
 
+  @PostMapping(
+    path = ["/veto"]
+  )
+  @ResponseStatus(ACCEPTED)
+  fun veto(
+    @RequestHeader("X-SPINNAKER-USER") user: String,
+    @RequestBody veto: EnvironmentArtifactVeto
```

Comment:
Since this is the only place we use this class (the one named in the singular), is it worth having it or could we just use `EnvironmentArtifactVetoes` and the UI/user will pass an array with a single version if needed?

I'm only asking because I got a bit confused when I saw the two classes looking almost identical in the model.

Reply:
The internal class has `DeliveryArtifact` on it, the external class has the (reference, type) properties by which users refer to a `DeliveryArtifact`.

Reply:
Ah, Ok! I missed that.